cmake/dependencies.cmake

```diff
@@ -0,0 +1,83 @@
+# Download dependencies by using FetchContent_Declare
+# Use FetchContent_MakeAvailable only in those code parts where the dependency is actually needed
+
+include(FetchContent)
+set(FETCHCONTENT_QUIET OFF)
+
+FetchContent_Declare(gtest
+    GIT_REPOSITORY https://github.com/google/googletest
+    GIT_TAG v1.13.0
+    GIT_SHALLOW ON
+    GIT_PROGRESS ON
+    FIND_PACKAGE_ARGS 1.13.0)
+
+# We need to make gtest available here, because benchmark depends on it
+FetchContent_MakeAvailable(gtest)
```

Comment:
We can do that also in test and benchmark, we do no need to download this everytime. Or iam wrong?

Reply:
Benchmark depends on gtest. My current solution is a bad solution because benchmark just must be configured differently.

Reply:
I just mean put `FetchContent_MakeAvailable(gtest)` before `FetchContent_MakeAvailable(benchmark)`

Reply:
Done.